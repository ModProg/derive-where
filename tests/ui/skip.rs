use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Debug)]
struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateSkipSeparate<T>(
	#[derive_where(skip)]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);

#[derive_where(Debug)]
struct OverridingSkip<T>(
	#[derive_where(skip)]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Debug)]
struct UnderridingSkip<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip)]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);

#[derive_where(Debug)]
struct DuplicateTraitSeparate<T>(
	#[derive_where(skip(Debug))]
	#[derive_where(skip(Debug))]
	PhantomData<T>,
);

#[derive_where(Clone; T)]
struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);

#[derive_where(Clone, Copy)]
struct SkipCloneWhileCopy<T>(#[derive_where(skip(Clone))] PhantomData<T>);

fn main() {}
