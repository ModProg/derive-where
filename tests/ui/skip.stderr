error: duplicate `skip` option
 --> tests/ui/skip.rs:6:50
  |
6 | struct DuplicateSkipSame<T>(#[derive_where(skip, skip)] PhantomData<T>);
  |                                                  ^^^^

error: duplicate `skip` option
  --> tests/ui/skip.rs:11:17
   |
11 |     #[derive_where(skip)]
   |                    ^^^^

error: empty attribute option found
  --> tests/ui/skip.rs:16:36
   |
16 | struct EmptySkip<T>(#[derive_where(skip())] PhantomData<T>);
   |                                    ^^^^^^

error: unexpected constraint on `skip` when unconstrained `skip` already used
  --> tests/ui/skip.rs:21:17
   |
21 |     #[derive_where(skip(Debug))]
   |                    ^^^^^^^^^^^

error: duplicate `skip` option
  --> tests/ui/skip.rs:28:17
   |
28 |     #[derive_where(skip)]
   |                    ^^^^

error: no trait that can be skipped is being implemented
  --> tests/ui/skip.rs:33:43
   |
33 | struct NoSupportedTrait<T>(#[derive_where(skip)] PhantomData<T>);
   |                                           ^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/skip.rs:36:57
   |
36 | struct DuplicateTraitSame<T>(#[derive_where(skip(Debug, Debug))] PhantomData<T>);
   |                                                         ^^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/skip.rs:41:22
   |
41 |     #[derive_where(skip(Debug))]
   |                         ^^^^^

error: trait to be skipped isn't being implemented
  --> tests/ui/skip.rs:46:50
   |
46 | struct MissingDeriveTrait<T>(#[derive_where(skip(Debug))] PhantomData<T>);
   |                                                  ^^^^^

error: Cannot skip `Clone` while deriving `Copy`
  --> tests/ui/skip.rs:49:50
   |
49 | struct SkipCloneWhileCopy<T>(#[derive_where(skip(Clone))] PhantomData<T>);
   |                                                  ^^^^^
